include/zephyr/bluetooth/l2cap.h

```diff
@@ -147,7 +147,12 @@ struct bt_l2cap_le_endpoint {
 	uint16_t				mtu;
 	/** Endpoint Maximum PDU payload Size */
 	uint16_t				mps;
-	/** Endpoint initial credits */
+	/** Endpoint initial credits
+	 *
+	 * The user should allocate enough credits to be able to transfer a full
+	 * SDU. 1 credit = 1 PDU of MPS length. Example:
+	 * SDU = 2000 bytes, MPS = 77 -> init_credits >= 26.
```

Comment:
This is not enough to ensure successful transport. The remote may choose to underutilize MPS. The comment we are removing below hinted at exactly this.

https://github.com/zephyrproject-rtos/zephyr/blob/bb6fa5707f77d72307008405f19dd83247405392/subsys/bluetooth/host/l2cap.c#L2460-L2463

We need a contract with the application:

When the channel connected event handlers complete, `bt_l2cap_chan_ops.alloc_seg` shall be ready give out `init_credits` net_bufs. When the net_buf is destructor returns, a net_buf must be available from `alloc_seg` again. 

`bt_l2cap_chan_ops.alloc_buf` shall be ready give out one buffer after..
 - channel connected event handlers complete
 - when the channel's `recv` returns other than `-EINPROGRESS`
 - when the channel's `bt_l2cap_chan_recv_complete` is invoked

Contract is valid until the 'disconnected' event.

Additionally: (deprecated) If `bt_l2cap_chan_ops.alloc_seg` is NULL, then `alloc_buf` must in addition give out the buffers demanded of `bt_l2cap_chan_ops.alloc_seg`.

We make sure the data in a `seg` is copied to the `buf`, and `seg` is destroyed before we accept more ACL data from that conn.